### PR TITLE
feat: Upgrade JAXB version - moving javax package to jakarta, close #319

### LINKED ISSUE #319: Montee Version du package xml et JAXB
Passage de javax.* à jakarta.*
Montée de version

### PR DESCRIPTION
**Pull request by gleizesDor**
- Passage de javax.* à jakarta.*

BREAKING CHANGE: Passage de javax.* à jakarta.*

### CODE REVIEW

File: sct-commons/pom.xml [hunk 1]
```diff
@@ -99,12 +99,6 @@
             <artifactId>jakarta.annotation-api</artifactId>
```
Comment (samirromdhani) [inline sct-commons/pom.xml:99]: This dependency is never used.  It is unaffected by the upgrade, so we can remove it.
> [!WARNING]
> [INFO] --- maven-dependency-plugin:3.3.0:analyze (default-cli) @ sct-commons ---
[WARNING] Used undeclared dependencies found:
......
[WARNING]    jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile
...

File: sct-commons/pom.xml [hunk 2]
```diff
@@ -35,20 +35,20 @@
...
+            <version>4.0.0</version>
         </dependency>
         <dependency>
-            <groupId>com.sun.xml.bind</groupId>
-            <artifactId>jaxb-impl</artifactId>
-            <version>2.3.3</version>
-            <scope>runtime</scope>
+            <groupId>org.glassfish.jaxb</groupId>
+            <artifactId>jaxb-runtime</artifactId>
+            <version>4.0.3</version>
```
[the remark this answers is not in the document]
Comment (gleizesDor) [reply sct-commons/pom.xml:51]: I only need them in the commons module, so I created 2 properties for the 2 versions but inside this module

File: sct-commons/pom.xml (hunk 1 above)
Comment (gleizesDor) [reply sct-commons/pom.xml:99]: ok

File: sct-commons/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (samirromdhani) [reply sct-commons/pom.xml]: It's just an announcement. `dependencyManagement` does not really add the dependencies. It is a way to centralize the management of dependencies.

**Review by samirromdhani — APPROVED** [review]

File: sct-commons/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (gleizesDor) [reply sct-commons/pom.xml]: yes, it is usefull for Managed dependencies; 
here I only need these in common module, so no need to manage them

**Pull request merged**